examples/palettization/dkm_palettization.py

```diff
@@ -0,0 +1,225 @@
+# -*- coding: utf-8 -*-
+"""
+.. _palettization_tutorial:
+
+Palettization Using Differentiable K-Means
+==========================================
+
+"""
+
+########################################################################
+# In this tutorial, you learn how to palettize a
+# network trained on `MNIST <http://yann.lecun.com/exdb/mnist/>`_ using
+# :py:class:`~.palettizer.DKMPalettizer`.
+#
+# Learn more about other palettization in the coremltools 
+# `Training-Time Palettization Documentation <https://coremltools.readme.io/v7.0/docs/training-time-palettization>`_.
+
+
+########################################################################
+# Defining the Network and Dataset
+# --------------------------------
+#
+# First, define your network:
+
+from collections import OrderedDict
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+
+def mnist_net(num_classes=10):
+    return nn.Sequential(OrderedDict([
+        ('conv1', nn.Conv2d(1, 32, 5, padding='same')),
+        ('relu1', nn.ReLU()),
+        ('pool1', nn.MaxPool2d(2, stride=2, padding=0)),
+        ('bn1', nn.BatchNorm2d(32, eps=0.001, momentum=0.01)),
+        ('conv2', nn.Conv2d(32, 64, 5, padding='same')),
+        ('relu2', nn.ReLU()),
+        ('pool2', nn.MaxPool2d(2, stride=2, padding=0)),
+        ('flatten', nn.Flatten()),
+        ('dense1', nn.Linear(3136, 1024)),
+        ('relu3', nn.ReLU()),
+        ('dropout', nn.Dropout(p=0.4)),
+        ('dense2', nn.Linear(1024, num_classes)),
+        ('softmax', nn.LogSoftmax())]))
+
+
+########################################################################
+# For training, use the MNIST dataset provided by
+# `PyTorch <https://pytorch.org/vision/stable/generated/torchvision.datasets.MNIST.html#mnist>`_.
+# Apply a very simple transformation to the input images to normalize them.
+
+import os
```

Comment:
I think you should just have all imports at the beginning of the file.

Reply:
Because this is not a python file, it's a tutorial written as a python file. 

Therefore, it's more likely to be executed as a iPython notebook, where the imports can be contained in the cell so that it is more easy to follow along rather than having a wall of imports in the beginning of the iPython notebook.